### PR TITLE
chore: update test script and turborepo dependency

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Updated `turborepo > 1.2.0` (to include the `--filter` flag [Changelog](https://github.com/vercel/turborepo/releases/tag/v1.2.0)).
- Added `--filter=[HEAD^1]` to our `test` script, to only test what has changed since the previous commit.

## 🧢 Checklist

- [ ] Documentation
- [x] Tests
- [x] Ready to be merged

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "build": "turbo run build --scope=next-auth --scope=@next-auth/* --no-deps",
     "lint": "turbo run lint --scope=!next-auth-docs",
     "lint:fix": "turbo run lint -- --fix",
-    "test": "turbo run test --concurrency=1 --scope=!@next-auth/pouchdb-adapter --scope=!next-auth-*",
+    "test": "turbo run test --concurrency=1 --scope=!@next-auth/pouchdb-adapter --scope=!next-auth-* --filter=[HEAD^1]",
```
Comment (ThangHuuVu) [inline package.json:18]: Since --scope is deprecated in 1.2.x, do you think we should remove all `--scope` in this PR too?

Comment (ndom91) [reply package.json:18]: Where did you see that? Not finding anything in their [changelog](https://github.com/vercel/turborepo/releases) :thinking: 

But yeah in that case lets remove it and/or find a replacement for it. The point of the two `--scope` arguments (as far as I understood it) was the following:

- To ignore v3 adapters (i.e. `pouchdb-adapter`)
- To not build when changes are only in `next-auth-app` or `next-auth-gatsby`, i.e. those kinds of demo apps/directories.

Comment (ThangHuuVu) [reply package.json:18]: Yeah, sorry I didn't link it (was on mobile 😄) https://turborepo.org/docs/core-concepts/scopes It's on their docs site
> The point of the two --scope arguments (as far as I understood it) was the following:

    To ignore v3 adapters (i.e. pouchdb-adapter)
    To not build when changes are only in next-auth-app or next-auth-gatsby, i.e. those kinds of demo apps/directories.

You got it right!

Comment (ThangHuuVu) [inline package.json:18]:
```suggestion
    "test": "turbo run test --concurrency=1 --filter=!@next-auth/pouchdb-adapter --filter=!next-auth-* --filter=[HEAD^1]",
```

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
...
     "setup": "turbo run setup",
     "dev": "yarn dev:app",
     "dev:app": "turbo run dev --parallel --no-deps --no-cache --scope=next-auth-app",
```
Comment (ThangHuuVu) [inline package.json:21]:
```suggestion
    "dev:app": "turbo run dev --parallel --no-deps --no-cache --filter=next-auth-app",
```

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "build": "turbo run build --scope=next-auth --scope=@next-auth/* --no-deps",
```
Comment (ThangHuuVu) [inline package.json:15]:
```suggestion
    "build": "turbo run build --filter=next-auth --filter=@next-auth/* --no-deps",
```

File: package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
...
     "lint": "turbo run lint --scope=!next-auth-docs",
```
Comment (ThangHuuVu) [inline package.json:16]:
```suggestion
    "lint": "turbo run lint --filter=!next-auth-docs",
```

**Review by ThangHuuVu — COMMENTED** [review]
I suggested replacing all `--scope` with `--filter` in this PR. I tested the commands locally, they should work just like before 🙏 (my suggestions are missing some command because I couldn't comment on them)

File: package.json [hunk 5]
```diff
@@ -12,13 +12,13 @@
   "repository": "https://github.com/nextauthjs/next-auth.git",
   "scripts": {
     "build:app": "turbo run build --scope=next-auth-app --include-dependencies",
```
Comment (ThangHuuVu) [inline package.json:14]:
```suggestion
    "build:app": "turbo run build --filter=next-auth-app --include-dependencies",
```

File: package.json [hunk 6]
```diff
@@ -12,13 +12,13 @@
...
+    "build": "turbo run build --filter=next-auth --filter=@next-auth/* --no-deps",
+    "lint": "turbo run lint --filter=!next-auth-docs",
     "lint:fix": "turbo run lint -- --fix",
-    "test": "turbo run test --concurrency=1 --scope=!@next-auth/pouchdb-adapter --scope=!next-auth-*",
+    "test": "turbo run test --concurrency=1 --filter=!@next-auth/pouchdb-adapter --filter=!next-auth-* --filter=[HEAD^1]",
     "setup": "turbo run setup",
     "dev": "yarn dev:app",
-    "dev:app": "turbo run dev --parallel --no-deps --no-cache --scope=next-auth-app",
+    "dev:app": "turbo run dev --parallel --no-deps --no-cache --filter=next-auth-app",
     "dev:docs": "turbo run dev --parallel --no-deps --no-cache --scope=next-auth-docs",
```
Comment (ThangHuuVu) [inline package.json:22]:
```suggestion
    "dev:docs": "turbo run dev --parallel --no-deps --no-cache --filter=next-auth-docs",
```

**Review by ThangHuuVu — COMMENTED** [review]
The last two comments 🙏

**Comment by ndom91** [comment]
@ThangHuuVu Awesome, thanks! So those were the last two invocations of `--scope`, right? Should be good to merge then :tada:

**Comment by ThangHuuVu** [comment]
@ndom91 yep, should be ready now 🎉

**Review by ThangHuuVu — APPROVED** [review]

**Comment by ndom91** [comment]
> @ndom91 yep, should be ready now 🎉 

Hmm did it work for you locally?

**Comment by ThangHuuVu** [comment]
@ndom91 yeah, let's just merge it 👍

**Pull request merged**